clients/src/main/java/io/strimzi/models/KafkaConsumerRecord.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.models;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.record.TimestampType;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The type Kafka consumer record.
+ */
+public record KafkaConsumerRecord(long timestamp, TimestampType timestampType, String topic, int partition, long offset,
+                                  String key, String payload, List<Map<String, String>> headers) {
+    /**
+     * Gets topic.
+     *
+     * @return the topic
+     */
+    public String getTopic() {
+        return topic;
+    }
+
+    /**
+     * Gets partition.
+     *
+     * @return the partition
+     */
+    public int getPartition() {
+        return partition;
+    }
+
+    /**
+     * Gets offset.
+     *
+     * @return the offset
+     */
+    public long getOffset() {
+        return offset;
+    }
+
+    /**
+     * Gets timestamp.
+     *
+     * @return the timestamp
+     */
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    /**
+     * Gets timestamp type.
+     *
+     * @return the timestamp type
+     */
+    public String getTimestampType() {
+        return timestampType.name;
+    }
+
+    /**
+     * Gets key.
+     *
+     * @return the key
+     */
```

Comment:
Why did you implement the getters and setters here? Don't records in Java automatically provide getter methods?

Reply:
That's for the object mapper I suppose. To create the JSON output.

Reply:
Nice catch! I did it for the reason commented by Lukas. I thought it was necessary for the JSON output, but testing it, the getters are correctly provided by Java and it worked fine.